### PR TITLE
u-boot: restore bootloader/firmware script

### PR DESCRIPTION
**Pull request by Kwiboo**
This PR restores the `bootloader/firmware` script that was removed in #4762 and ensure `crust` is rebuilt for each unique release image. The `bootloader/firmware` script is needed to complete #4591.
Source of `bootloader/firmware` is also moved until after `make mrproper` to avoid issue with u-boot clean removing i.e. `bl31.elf` on rockchip platform.

It also decouple `crust` from the `u-boot` package and is no longer executing `scripts/uboot_helper` when `u-boot` `package.mk` is sourced.
~~The `linux` package is still executing `scripts/uboot_helper` when `linux` `package.mk` is sourced and is probably something that should be looked at in a separate PR.~~
`linux` package was changed to use `crust` patch dir when `UBOOT_FIRMWARE` contains `crust`.

Note: I have not tried a full Allwinner image re-build and have not done any runtime testing.

cc @dhewg @jernejsk

### CODE REVIEW

**Comment by jernejsk** [comment]
Changes seems reasonable. Note that Crust is currently supported only with H6. Can you remove it from other device options file? @dhewg can you test it on your setup?

**Comment by dhewg** [comment]
Thanks for this, looks nice and cleaner than my approach!

Didn't yet build- or runtime-test yet though, will report back once I'll get to it

**Comment by jernejsk** [comment]
sorry, I missed other crust targets set. @Kwiboo disregard my request for change

File: projects/Allwinner/bootloader/firmware [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2019-present Team LibreELEC (https://libreelec.tv)
+
+[ -n "$ATF_PLATFORM" ] && cp -av $(get_install_dir atf)/usr/share/bootloader/bl31.bin .
+
+CRUST_CONFIG=$($ROOT/$SCRIPTS/uboot_helper $PROJECT $DEVICE $UBOOT_SYSTEM crust_config)
+[ -n "$CRUST_CONFIG" ] && cp -av $(get_install_dir crust)/usr/share/bootloader/scp.bin .
```
Comment (jernejsk) [inline projects/Allwinner/bootloader/firmware:7]: Can you wrap this into proper if then fi? In current form it returns non-zero return code if there is no crust_config and that stops U-Boot build with error.

**Review by jernejsk — CHANGES_REQUESTED** [review]

File: projects/Allwinner/bootloader/firmware (hunk 1 above)
Comment (jernejsk) [reply projects/Allwinner/bootloader/firmware:7]: I guess adding `exit 0` would be cleaner

Comment (Kwiboo) [reply projects/Allwinner/bootloader/firmware:7]: I added a `exit 0` at end of the file, not sure this is what you had in mind :-)

Comment (jernejsk) [reply projects/Allwinner/bootloader/firmware:7]: it is, thanks!

**Review by jernejsk — APPROVED** [review]

**Pull request merged**